lib/brewdler/dsl.rb

```diff
@@ -1,23 +1,47 @@
 module Brewdler
   class Dsl
+    class Entry
+      attr_reader :type, :name, :extra
+
+      def initialize(type, name, extra={})
+        @type = type
+        @name = name
+        @extra = extra
+      end
+    end
+
+    attr_reader :entries
+
     def initialize(input)
       @input = input
+      @entries = []
     end
 
     def process
       instance_eval(@input)
+      self
+    end
+
+    def install
+      @entries.each do |entry|
+        case entry.type
+        when :brew then Brewdler::BrewInstaller.install(entry.name, entry.extra[:options])
+        when :cask then Brewdler::CaskInstaller.install(entry.name)
+        when :repo then Brewdler::RepoInstaller.install(entry.name)
+        end
+      end
     end
 
     def brew(name, options={})
-      Brewdler::BrewInstaller.install(name, options)
+      @entries << Entry.new(:brew, name, {:options => options})
```

Comment:
Here, I store `options` as a sub hash inside `extra`. The reason I use `extra` rather than `options` is I want to make it easier to extend for possible need in the future. By design, we can store anything in `extra`.